oat-sa/tao-core-sdk-fe
# [INTEGRATION] Fix/ADF-1166/ESlint errors

# This is an integration branch. It is very big, don't try reviewing it, unless you are brave enough... Actually, the changes have been split into several PR, so this one will never get merged 😄 

### Related to: https://oat-sa.atlassian.net/browse/ADF-1166

### Summary

Integration branch for the elimination of ESLint errors and warnings in `tao-core-sdk`.

**Note:** as the changes are pretty heavy, I split them all into several pull requests:
- https://github.com/oat-sa/tao-core-sdk-fe/pull/153 (this one is the first step for tao-core-sdk)
- https://github.com/oat-sa/tao-core-sdk-fe/pull/154 (eslint fixes)
- https://github.com/oat-sa/tao-core-sdk-fe/pull/155 (eslint fixes)
- https://github.com/oat-sa/tao-core-sdk-fe/pull/156 (eslint fixes, the biggest one, sorry)
- https://github.com/oat-sa/tao-core-sdk-fe/pull/157 (eslint fixes, the second biggest)
- https://github.com/oat-sa/tao-core-sdk-fe/pull/158 (eslint fixes)

### Details

What has been made:
- fix errors that can be automated using `eslint --fix`, then adjust the changes in case by case (mostly string literals)
- fix the wrong annotations (missing/wrong return or param definition)
- fix the more complex cases with some refactoring (rest operator instead of `.apply()`)
- replace `var` by `const` or `let` (but not in all places, only in the touched files when it was not adding too much noise)
- apply object shorthand instead of duplicated function names
- apply arrow functions here and there when it helped, but not systematically to reduce the noise

### How to test
- checkout and install the source code `npm i`
- run the linter: `npm run lint:src`, errors and warnings should have gone.
- run the linter: `npm run lint:test`, errors and warnings should have gone.
- run the unit tests: `npm run test`
- install with other repositories, using the following scheme: `"@oat-sa/tao-core-sdk": "https://github.com/oat-sa/tao-core-sdk-fe.git#fix/ADF-1166/eslint-errors",`, then checking the unit tests and playing with TAO
